src/include/storage/wal/wal.h

```diff
@@ -25,16 +24,13 @@ struct SequenceRollbackData;
 } // namespace catalog
 
 namespace storage {
-
-using lock_t = std::unique_lock<std::mutex>;
-
 class WALReplayer;
 class WAL {
     friend class WALReplayer;
 
 public:
-    WAL(const std::string& directory, bool readOnly, BufferManager& bufferManager,
-        common::VirtualFileSystem* vfs, main::ClientContext* context);
+    WAL(const std::string& directory, bool readOnly, common::VirtualFileSystem* vfs,
```

Comment:
if u are passing around clientContext, u probably don't need to pass around vfs

Reply:
This is something I need to discuss with @acquamarin regarding to how vfs should make use of ClientContext. We are passing both in several different places, which should refactored altogether if possible.